### ISSUE TITLE
BUG: non-functional pd.Series -> DNAFASTAFormat transformer in feature_data subpackage

### ISSUE THREAD

**Issue by jakereps** [qiime2/q2-types#149]
I've included an example code block showing the error at the bottom of this issue. Pulling the `_16` transformer into a local function and adding a `skbio.DNA` wrapper around the sequence string allowed it to partially work as expected (sans header ids).

```python
In [11]: def _16(data: pd.Series) -> DNAFASTAFormat:
    ...:     ff = DNAFASTAFormat()
    ...:     with ff.open() as f:
    ...:         for sequence in data:
    ...:             skbio.io.write(skbio.DNA(sequence), format='fasta', into=f)
    ...:     return ff
    ...: 

In [12]: f = _16(features.loc[data.columns, 'DenoisedSequenceVariant'])

In [13]: !head {f.path}
>
GCGAGCGT...
>
GCAAGCGT...
>
GCAAGCGT...
>
GCAAGCGT...
>
GCAAGCGT...
```

The funny(/ironic?) part is that it's the only transformer in the sub-package without any tests.
```python
In [6]: qiime2.Artifact.import_data('FeatureData[Sequence]', features.loc[data.columns, 'DenoisedSequenceVariant'])
---------------------------------------------------------------------------
UnrecognizedFormatError                   Traceback (most recent call last)
<ipython-input-6-f8fdc74db9db> in <module>()
----> 1 qiime2.Artifact.import_data('FeatureData[Sequence]', features.loc[data.columns, 'DenoisedSequenceVariant'])

~/dev/mc3/envs/[redacted]/lib/python3.5/site-packages/qiime2/sdk/result.py in import_data(cls, type, view, view_type)
    190 
    191         provenance_capture = archive.ImportProvenanceCapture(format_, md5sums)
--> 192         return cls._from_view(type_, view, view_type, provenance_capture)
    193 
    194     @classmethod

~/dev/mc3/envs/[redacted]/lib/python3.5/site-packages/qiime2/sdk/result.py in _from_view(cls, type, view, view_type, provenance_capture)
    215         transformation = from_type.make_transformation(to_type,
    216                                                        recorder=recorder)
--> 217         result = transformation(view)
    218 
    219         artifact = cls.__new__(cls)

~/dev/mc3/envs/[redacted]/lib/python3.5/site-packages/qiime2/core/transform.py in transformation(view)
     57             self.validate(view)
     58 
---> 59             new_view = transformer(view)
     60 
     61             new_view = other.coerce_view(new_view)

~/dev/mc3/envs/[redacted]/lib/python3.5/site-packages/qiime2/core/transform.py in wrapped(view)
    205         def wrapped(view):
    206             new_view = self._view_type()
--> 207             file_view = transformer(view)
    208             if transformer is not identity_transformer:
    209                 self.set_user_owned(file_view, False)

~/dev/mc3/envs/[redacted]/lib/python3.5/site-packages/q2_types/feature_data/_transformer.py in _16(data)
    339     with ff.open() as f:
    340         for sequence in data:
--> 341             skbio.io.write(sequence, format='fasta', into=f)
    342     return ff
    343 

~/dev/mc3/envs/[redacted]/lib/python3.5/site-packages/skbio/io/registry.py in write(obj, format, into, **kwargs)
   1164 @wraps(IORegistry.write)
   1165 def write(obj, format, into, **kwargs):
-> 1166     return io_registry.write(obj, format, into, **kwargs)
   1167 
   1168 

~/dev/mc3/envs/[redacted]/lib/python3.5/site-packages/skbio/io/registry.py in write(self, obj, format, into, **kwargs)
    615             raise UnrecognizedFormatError(
    616                 "Cannot write %r into %r, no %s writer found." %
--> 617                 (format, into, obj.__class__.__name__))
    618 
    619         writer(obj, into, **kwargs)

UnrecognizedFormatError: Cannot write 'fasta' into <_io.TextIOWrapper name='/tmp/q2-DNAFASTAFormat-ohosxny8' mode='r+' encoding='utf8'>, no str writer found.

In [7]: 
```

I can work on this if necessary!

**Comment by jairideout** [comment]
Nice catch! One option is to remove this transformer outright (since apparently it never worked), similar to what we did with the broken sequence iterators in q2-types. Is this a transformer that you'd like to have working, or is removing it okay?

**Comment by jairideout** [comment]
Apologies, I just saw that you already have an open PR to fix it! I'll add it to the 2017.9 release sprint. Thanks for fixing it @jakereps!
